libs/coin-evm/src/adapters/ledger.ts

```diff
@@ -222,3 +224,68 @@ export const ledgerERC1155EventToOperations = (
     );
   });
 };
+
+/**
+ * Adapter to convert an internal transaction
+ * on Ledger explorers into LL Operations
+ */
+export const ledgerInternalTransactionToOperations = (
+  coinOperation: Operation,
+  action: LedgerExplorerInternalTransaction,
+  index = 0,
+): Operation[] => {
+  const { hash, blockHeight, blockHash, date, accountId } = coinOperation;
+  const { xpubOrAddress: address } = decodeAccountId(accountId);
+  const from = safeEncodeEIP55(action.from);
+  const to = safeEncodeEIP55(action.to);
+  const checksummedAddress = eip55.encode(address);
+  const hasFailed = !!action.error; // AFAIK this is not working, all actions contain error = null even when it reverted
+  const value = new BigNumber(action.value);
+  const types: OperationType[] = [];
+
+  // Ledger explorers are indexing the first `CALL` opcode of a smart contract transaction as an
+  // internal transaction which is wrong. Only children `CALL` opcode should be indexed,
+  // therefore we need to filter those "actions" to prevent duplicating ops.
```

Comment:
Has this been discussed with backend team?

Reply:
Nope, but AFAICT, it has always been like this, we were already doing this shenanigan with the `ethereum` family (and we were doing it incorrectly at the time, but that's not important 🤷‍♂️)